tools/PI/DevHome.PI/ViewModels/ClipboardMonitorControlViewModel.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.PI.Models;
+
+namespace DevHome.PI.ViewModels;
+
+public partial class ClipboardMonitorControlViewModel : ObservableObject
+{
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    [ObservableProperty]
+    private string _clipboardContentsHex = string.Empty;
+
+    [ObservableProperty]
+    private string _clipboardContentsDec = string.Empty;
+
+    [ObservableProperty]
+    private string _clipboardContentsCode = string.Empty;
+
+    [ObservableProperty]
+    private string? _clipboardContentsHelp;
+
+    private bool _listenForClipboardChanges = true;
+
+    public ClipboardMonitorControlViewModel()
+    {
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+        ClipboardMonitor.Instance.PropertyChanged += Clipboard_PropertyChanged;
+
+        PopulateClipboardData(ClipboardMonitor.Instance.Contents);
+    }
+
+    private void Clipboard_PropertyChanged(object? sender, PropertyChangedEventArgs e)
+    {
+        var clipboardContents = ClipboardMonitor.Instance.Contents;
+        _dispatcher.TryEnqueue(() =>
+        {
+            if (_listenForClipboardChanges)
+            {
+                PopulateClipboardData(clipboardContents);
+            }
+        });
+    }
+
+    private void PopulateClipboardData(ClipboardContents clipboardContents)
+    {
+        ClipboardContentsHex = clipboardContents.Hex;
+        ClipboardContentsDec = clipboardContents.Dec;
+        ClipboardContentsCode = clipboardContents.Code;
+        ClipboardContentsHelp = string.IsNullOrEmpty(clipboardContents.Help) ? null : clipboardContents.Help;
```

Comment:
If the string is null or empty, assign `null`.  Is this to catch the empty case?

Reply:
Yep. An empty string gives you an empty tooltip (instead of no tool tip at all). I'll add a comment.